haystack/preview/document_stores/memory/document_store.py

```diff
@@ -142,3 +157,92 @@ def delete_documents(self, document_ids: List[str]) -> None:
+        return self._tokenizer
+
+    @bm25_tokenization_regex.setter
+    def bm25_tokenization_regex(self, regex_string: str):
+        """
+        Setter for the regular expression used for tokenization in the BM25 algorithm.
+
+        :param regex_string: The regex string to be used for tokenization.
+        """
+        self._tokenizer = re.compile(regex_string).findall
+
+    @property
+    def bm25_algorithm(self):
+        """
+        The BM25 algorithm used for retrieval.
+
+        :getter: Returns the BM25 algorithm class used for retrieval.
+        """
+        return self._bm25_class
+
+    @bm25_algorithm.setter
+    def bm25_algorithm(self, algorithm: str):
+        """
+        Setter for the BM25 algorithm used for retrieval.
+
+        :param algorithm: The name of the BM25 algorithm to be used for retrieval.
+        The following algorithms are available:
+         - BM25Okapi (default)
+         - BM25L
+         - BM25Plus
+        """
+        algorithm_class = getattr(rank_bm25, algorithm)
+        if algorithm_class is None:
+            raise ValueError(f"BM25 algorithm '{algorithm}' not found.")
+        self._bm25_class = algorithm_class
+
+    def bm25_retrieval(self, query: str, top_k: int = 10, scale_score: bool = True) -> List[Document]:
+        """
+        Retrieves documents that are most relevant to the query using BM25 algorithm.
+
+        :param query: The query string.
+        :param top_k: The number of top documents to retrieve. Default is 10.
+        :param scale_score: Whether to scale the scores of the retrieved documents. Default is True.
+        :return: A list of the top 'k' documents most relevant to the query.
+        """
+
+        all_documents = self.filter_documents(filters={"content_type": ["text", "table"]})
+        lower_case_documents = []
+        for doc in all_documents:
+            if doc.content_type == "text":
+                lower_case_documents.append(doc.content.lower())
+            elif doc.content_type == "table":
+                if isinstance(doc.content, pd.DataFrame):
+                    lower_case_documents.append(doc.content.astype(str).to_csv(index=False).lower())
+
+        tokenized_corpus = [
+            self.bm25_tokenization_regex(doc)
+            for doc in tqdm(lower_case_documents, unit=" docs", desc="Updating BM25 representation...")
+        ]
+        if len(tokenized_corpus) == 0:
+            logger.warning("No documents found for BM25 retrieval. Returning empty list.")
+            return []
+
+        self.bm25 = self.bm25_algorithm(tokenized_corpus, **self.bm25_parameters)
+
+        tokenized_query = self.bm25_tokenization_regex(query.lower())
+        docs_scores = self.bm25.get_scores(tokenized_query)
+        if scale_score is True:
+            # scaling probability from BM25
+            docs_scores = [float(expit(np.asarray(score / 8))) for score in docs_scores]
+        top_docs_positions = np.argsort(docs_scores)[::-1][:top_k]
```

Comment:
This is equivalent and we avoid a reverse.

```suggestion
        top_docs_positions = np.argsort(docs_scores)[-top_k:]
```

Reply:
I just gave it a try and it worked with no issues, are you there weren't failing for some other reasons? 🤔

Reply:
I also tested it and it works, so I updated the PR